snuba/subscriptions/data.py

```diff
@@ -161,18 +169,21 @@ def build_request(
             dataset,
             timer,
             referrer,
-            [
-                self.entity_subscription.validate_query,
-                partial(self.add_conditions, timestamp, offset),
-            ],
+            custom_processing,
         )
         return request
 
     @classmethod
     def from_dict(
         cls, data: Mapping[str, Any], entity_key: EntityKey
     ) -> SubscriptionData:
-        entity_subscription = get_entity_subscription(entity_key)(data_dict=data)
+        entity_subscription = get_entity(entity_key).get_entity_subscription()
+        assert entity_subscription is not None
+
+        metadata = {}
+        for key in data.keys():
+            if key not in SUBSCRIPTION_DATA_PAYLOAD_KEYS:
+                metadata[key] = data[key]
```

Comment:
This seems too complicated and now you are keeping these keys in two places.

Instead of doing `data["project_id"]` etc below why don't you just `data.pop("project_id")`. Everything remaining in `data` after you've popped the parts needed for SubscriptionData will be your metadata.